src/modules/manaFiatGateway/sagas.ts

```diff
@@ -43,11 +44,18 @@ export function createManaFiatGatewaysSaga(config: ManaFiatGatewaySagasConfig) {
     yield takeEvery(purchaseEventsChannel, handlePurchaseChannelEvent)
 
     function* handlePurchaseChannelEvent(action: { purchase: Purchase }) {
-      yield put(setPurchase(action.purchase))
+      const { purchase } = action
+      yield put(setPurchase(purchase))
+      yield call(updateBalanceOnPurchaseCompletion, purchase)
     }
   }
 }
 
+function* updateBalanceOnPurchaseCompletion(purchase: Purchase) {
```

Comment:
What do you think if we subscribe this generator to each `SET_PURCHASE` event? The 2 times that you're calling it here are both after the `yield put(setPurchase(purchase))` event.
It'd be something like:
```
yield takeEvery(purchaseEventsChannel, updateBalanceOnPurchaseCompletion)
```
and the `updateBalanceOnPurchaseCompletion` would recieve the action so then we can grab the purchase from the payload.
I think looks a bit cleaner, what do you think?